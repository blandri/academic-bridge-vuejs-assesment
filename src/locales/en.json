{
  "board": "Board",
  "delete": "Delete",
  "edit": "Edit",
  "search": "Search",
  "done": "Done",
  "underReview": "Under review",
  "doing": "Doing",
  "toDo": "To do",
  "add": "Add",
  "a": "a",
  "new": "new",
  "card": "card",
  "upload": "Upload",
  "to": "to",
  "title": "title"
}
