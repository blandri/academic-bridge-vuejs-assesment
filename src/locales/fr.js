export default {
  translation: {
    board: 'Tableaux',
    delete: 'Supprime',
    edit: 'Edit',
    search: 'Chercher',
    done: 'Complet',
    underReview: 'Under review',
    doing: 'Doing',
    toDo: 'To do',
    add: 'Ajouter',
    a: 'un',
    new: 'nouveaux',
    card: 'carte',
    upload: 'Upload'
  }
}
