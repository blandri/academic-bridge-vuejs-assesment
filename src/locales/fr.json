{
  "board": "Tableaux",
  "delete": "Supprime",
  "edit": "Modifier",
  "search": "Chercher",
  "done": "Complet",
  "underReview": "Revoir",
  "doing": "En cours",
  "toDo": "A faire",
  "add": "Ajouter",
  "a": "un",
  "new": "nouveaux",
  "card": "carte",
  "upload": "Telecharge",
  "to": "a",
  "title": "titre"
}
