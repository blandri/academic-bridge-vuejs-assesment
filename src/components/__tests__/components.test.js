import { mount } from "@vue/test-utils"
import { createPinia, setActivePinia } from "pinia"
import { beforeEach, describe, expect, test } from "vitest"
import DropDownItemVue from "../DropDownItem.vue"
import EditModalVue from "../EditModal.vue"
import { createI18n } from "vue-i18n"
import en from '../../locales/en.json'
import fr from '../../locales/fr.json'

describe('Components test', () => {
    let i18n;
    beforeEach(() => {
      setActivePinia(createPinia())

      i18n = createI18n({
        messages: {
           en,
           fr
        }
     });

    })
  
    test('test drop down item', async () => {
        const wrapper = mount(DropDownItemVue)
        expect(wrapper.text()).toMatchSnapshot()
    })

    test('test editing modal', async () => {
        const wrapper = mount(EditModalVue, {
            global: {
               plugins: [i18n]
            }
         })
        expect(wrapper.text()).toMatchSnapshot()
    })
})