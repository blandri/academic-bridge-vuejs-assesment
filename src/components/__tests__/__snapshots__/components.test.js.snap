// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components test > test drop down item 1`] = `"Barone LLC."`;

exports[`Components test > test editing modal 1`] = `"Edit todo card Upload Image Edit"`;

exports[`Components test > test status item 1`] = `"Edit todo card Upload Image Edit"`;

exports[`TodoList Store > create todo item 1`] = `"Barone LLC."`;
